client/extensions/woocommerce/state/wc-api/products/actions.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * Internal dependencies
+ */
+import wp from 'lib/wp';
+import { error } from '../actions';
+import {
+	WOOCOMMERCE_API_CREATE_PRODUCT,
+	WOOCOMMERCE_API_CREATE_PRODUCT_SUCCESS,
+} from '../../action-types';
+
+export function createProduct( siteId, product ) {
+	return ( dispatch ) => {
+		const createAction = {
+			type: WOOCOMMERCE_API_CREATE_PRODUCT,
+			payload: { siteId, product },
+		};
+
+		dispatch( createAction );
+
+		// Filter out any id we might have.
+		const { id, ...productData } = product;
+
+		if ( typeof id === 'number' ) {
+			dispatch( error( siteId, createAction, {
+				message: 'Attempting to create a product which already has a valid id.',
+				product,
+			} ) );
+			return;
+		}
+
+		const jetpackProps = { path: `/jetpack-blogs/${ siteId }/rest-api/` };
+		const httpProps = {
+			path: '/wc/v2/products',
+			body: JSON.stringify( productData ),
+			json: true,
+		};
+
+		return wp.req.post( jetpackProps, httpProps )
```

Comment:
I just recently added a data layer middleware for extensions. I plan on using it after some changes with the data layer go through, so the thunks are just temporary. But I did just add a couple `// TODO:` comments to reflect that. Thanks!